npc/events/christmas_2006.txt

```diff
@@ -0,0 +1,4165 @@
+		end;
+
+	OnTimer11000:
+		announce "Never remember the sad memory!!!.", bc_map, 0x66FFCC;
+		end;
+
+	OnTimer13000:
+		announce "...I am the happy single soldier!!!", bc_map, 0x66FFCC;
+		end;
+
+	OnTimer180000:
+		killmonster "prt_fild05", "Enjoy#06xmas2::OnMobDead";
+		announce "To all my single soldier friends~Merry christmas.", bc_map, 0x66FFCC;
+		donpcevent "Enjoy#06xmas2::OnStop";
+		end;
+
+	OnMobDead:
+		end;
+}
+
+prt_fild05,97,164,5	script	Happymerry#06xmas_02	4_M_LGTMAN,{
+	end;
+
+	OnInit:
+		disablenpc "Happymerry#06xmas_02";
+		end;
+
+	OnEnable:
+		enablenpc "Happymerry#06xmas_02";
+		emotion ET_GO;
+		end;
+}
+
+prt_fild05,103,164,4	script	Christ#06xmas_02	4_M_PECOKNIGHT,{
+	end;
+
+	OnInit:
+		disablenpc "Christ#06xmas_02";
+		end;
+
+	OnEnable:
+		enablenpc "Christ#06xmas_02";
+		emotion ET_GO;
+		end;
+}
+
+prt_fild05,103,159,1	script	Mas#06xmas_02	4_M_ALCHE_A,{
+	end;
+
+	OnInit:
+		disablenpc "Mas#06xmas_02";
+		end;
+
+	OnEnable:
+		enablenpc "Mas#06xmas_02";
+		emotion ET_GO;
+		end;
+}
+
+prt_fild05,97,159,7	script	Event#06xmas_02	4_M_SITDOWN,{
+	end;
+
+	OnInit:
+		disablenpc "Event#06xmas_02";
+		end;
+
+	OnEnable:
+		enablenpc "Event#06xmas_02";
+		emotion ET_GO;
+		end;
+}
+
+prt_fild05,188,177,4	script	Happymerry#06xmas01	4_M_LGTMAN,{
+	if (05christ_solo == 9) {
+		mes "[Happymerry]";
+		mes "God has betrayed me again!!Even Holgren!!!";
+		emotion ET_CRY;
+		next;
+		mes "["+ strcharinfo(0) +"]";
+		mes "... Are you Happymerry?";
+		next;
+		mes "[Happymerry]";
+		mes "Go head and laugh at me!You don't understand why I get dumped on christmas day!!";
+		next;
+		mes "["+ strcharinfo(0) +"]";
+		mes "Happymerry! Enjoy said...";
+		next;
+		mes "-I start stop talking and look around to see if anyone is listening.-";
+		next;
+		mes "["+ strcharinfo(0) +"]";
+		mes "Didn't Enjoy leave any clue for you?";
+		next;
+		mes "[Happymerry]";
+		mes "...!";
+		mes "Do you know Enjoy?";
+		mes "Actually, Enjoy told me to give a clue to someone who come to me talk about him.";
+		next;
+		mes "[Happymerry]";
+		mes "^0000ffGarden^000000!";
+		next;
+		mes "["+ strcharinfo(0) +"]";
+		mes "...What?What did you say?";
+		next;
+		mes "[Happymerry]";
+		mes "^0000ffGarden^000000!";
+		next;
+		mes "[Happymerry]";
+		mes "I'm sorry, that's all I know.Go ask other members.";
+		next;
+		mes "[Happymerry]";
+		mes "Well then, I'll meet you 'there'.";
+		05christ_solo = 10;
+		close;
+	} else if (05christ_solo > 9) {
+		mes "[Happymerry]";
+		mes "^0000ffGarden^000000!";
+		next;
+		mes "[Happymerry]";
+		mes "I'm sorry, that's all I know.Go ask other members.";
+		next;
+		mes "[Happymerry]";
+		mes "Well then, I'll meet you 'there'.";
+		close;
+	} else {
+		mes "[Happymerry]";
+		mes "God has betrayed me again!!Even Holgren!!!";
+		emotion ET_CRY;
+		close;
+	}
+}
+
+prontera,62,339,3	script	Christ#06xmas_01	4_M_PECOKNIGHT,{
+	if (05christ_solo == 10) {
+		mes "[Christ]";
```

Comment:
`.@name$ = strcharinfo(0);` to avoid the redundant command call

Reply:
Applied suggested fix.